src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -647,6 +651,30 @@ public void searchWorkOrder()
         }
     }
 
+    @Override
+    public void onDestroyed()
+    {
+        super.onDestroyed();
+        final List <Map.Entry<BlockPos, IBuilding>> guardbuildings = this.colony.getBuildingManager().getBuildings().entrySet()
+          .stream()
+          .filter(entry ->
```

Comment:
this seems like a huge overkill. Just iterate over all buildings and do .setMinePos(null) when the building is a guardtower and it has the mine set already. No need to collect them first and then iterate them again.

Reply:
we could also simply let the guards deal with this, they check buildingmanager.getBuilding(minerPos) != null and if that fails they switch out of the miner protection mode

Reply:
True, check if the building at that pos is a minerbuilding, and if not (null or other building) they switch the mode out.